File: 3945-digit-frequency-score/3945-digit-frequency-score.py
class Solution:
    def digitFrequencyScore(self, n: int) -> int:
        s=set()
        num=str(n)
        sum=0
        for i in num:
            if i not in s:
                sum+=(num.count(i)*(int)(i))
                s.add(i)
        return sum

File: 3945-digit-frequency-score/3945-digit-frequency-score.java
class Solution {
    public int digitFrequencyScore(int n) {
        int arr[]=new int[10];
        int val=n;
        while(val>0){
            int v=val%10;
            arr[v]++;
            val/=10;
        }
        int sum=0;
        for(int i=0; i<10; i++){
            sum+=(arr[i]*i);
        }
        return sum;
    }
}